src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8167,8 +8216,13 @@ void checkWellKnown(WellKnownMember member)
                             property.OriginalDefinition is { ParameterCount: 1 } original &&
                             original.Parameters[0] is { Type: { SpecialType: SpecialType.System_Int32 }, RefKind: RefKind.None })
                         {
-                            // note: implicit copy check on the indexer accessor happens in CheckPropertyValueKind
-                            patternSymbol = property;
+                            var intPlaceholder = new BoundIndexOrRangeIndexerPatternValuePlaceholder(syntax, Compilation.GetSpecialType(SpecialType.System_Int32)) { WasCompilerGenerated = true };
```

Comment:
>Compilation.GetSpecialType(SpecialType.System_Int32) [](http://example.com/codeflow?start=109&length=52)

Please use helper from the binder that reports use-site diagnostics, or extract the type from the corresponding parameter symbol.